src/view/selection.js

```diff
@@ -461,6 +462,25 @@ export default class Selection {
 	}
 
 	/**
+	 * Returns selected element. {@link engine.view.Element Element} is considered as selected if there is only
+	 * one range in selection, and that range is placed exactly on one element.
+	 * Returns `null` if there is no selected element.
+	 *
+	 * @return {engine.view.Element|null}
+	 */
+	getSelectedElement() {
+		if ( this.rangeCount == 1 ) {
+			const range = this.getFirstRange();
+			const nodeAfterStart = range.start.nodeAfter;
+			const nodeBeforeEnd = range.end.nodeBefore;
+
+			return nodeAfterStart instanceof Element && nodeAfterStart == nodeBeforeEnd ? nodeAfterStart : null;
```

Comment:
I'd use some parentheses for readability. It took me a couple of secs to parse this.